homeassistant/components/network/network.py

```diff
@@ -0,0 +1,237 @@
+"""Network helper class for the network integration."""
+from __future__ import annotations
+
+from ipaddress import ip_address
+import logging
+from typing import Any, Iterable, cast
+
+import ifaddr
+from pyroute2 import IPRoute
+
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.storage import Store
+
+from .const import (
+    ATTR_CONFIGURED_ADAPTERS,
+    DEFAULT_CONFIGURED_ADAPTERS,
+    NETWORK_CONFIG_SCHEMA,
+    STORAGE_KEY,
+    STORAGE_VERSION,
+)
+from .models import Adapter, IPv4ConfiguredAddress, IPv6ConfiguredAddress
+
+MDNS_TARGET_IP = "224.0.0.251"
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+ZC_CONF_DEFAULT_INTERFACE = (
+    "default_interface"  # cannot import from zeroconf due to circular dep
+)
+
+
+class Network:
+    """Network helper class for the network integration."""
+
+    def __init__(self, hass: HomeAssistant) -> None:
+        """Initialize the Network class."""
+        self.hass: HomeAssistant = hass
+        self._store: Store = hass.helpers.storage.Store(STORAGE_VERSION, STORAGE_KEY)
```

Comment:
These are automatically inferred.
```suggestion
        self.hass = hass
        self._store = hass.helpers.storage.Store(STORAGE_VERSION, STORAGE_KEY)
```